R/fimsframe.R

```diff
@@ -569,7 +569,35 @@ methods::setValidity(
     }
 
     # TODO: Add checks for other slots
+    # Add validity check for types
+    allowed_types <- c(
+  "landings", "index", "age_comp", "length_comp",
+  "weight-at-age", "age-to-length-conversion"
+)
+present_types <- unique(object@data$type)
```

Comment:
Could you change `object@data$type` to `object@data[["type"]]`? In FIMS, we use double brackets with quoted names instead of the dollar sign to avoid partial matching.